### ISSUE TITLE
Extending ParsePushBroadcastReceiver

### ISSUE THREAD

**Issue by ngocphamm** [ParsePlatform/Parse-SDK-dotNET#152]
Hi,

I'm trying to extend the ParsePushBroadcastReceiver so I can have some custom data parsed from the notification Intent Extras.

According to [this](https://parse.com/docs/android/guide#push-notifications-receiving-pushes), I should be able to extend the class. However, in C#, the class is `public sealed`. How am I supposed to do the subclass here, then?

I apologize if this appears to be a basic programming question, and thanks in advance.

**Comment by richardjrossiii** [comment]
Yep, this is definitely a bug, considering our official .NET guide says you should be able to subclass.

I'll get a fix out soon.

https://parse.com/tutorials/android-push-notifications

**Comment by ngocphamm** [comment]
Oh nice. Thanks for your response @richardjrossiii. Looking forward to an update!

**Comment by sw-ngoc** [comment]
I couldn't wait for the fix to come out, so I get this project (with your fix to this issue) and build the dll myself. However, I realize that even that wouldn't help me anyway.

This is the class in Java
https://github.com/ParsePlatform/Parse-SDK-Android/blob/master/Parse%2Fsrc%2Fmain%2Fjava%2Fcom%2Fparse%2FParsePushBroadcastReceiver.java

And this is the one in .NET
https://github.com/ParsePlatform/Parse-SDK-dotNET/blob/master/Parse%2FPublic%2FAndroid%2FParsePushBroadcastReceiver.cs

The thing is the .NET class doesn't have any of the methods like in Java to extend from. It calls the ParsePushService, and probably handle the receipt of notification over there, but I don't really know further how to deal with that.

Can you please confirm that there's a way to customize the way remote notification is received on Android, with Xamarin?

**Comment by ngocphamm** [comment]
I couldn't wait for the fix to come out, so I get this project (with your fix to this issue) and build the dll myself. However, I realize that even that wouldn't help me anyway.

This is the class in Java
https://github.com/ParsePlatform/Parse-SDK-Android/blob/master/Parse%2Fsrc%2Fmain%2Fjava%2Fcom%2Fparse%2FParsePushBroadcastReceiver.java

And this is the one in .NET
https://github.com/ParsePlatform/Parse-SDK-dotNET/blob/master/Parse%2FPublic%2FAndroid%2FParsePushBroadcastReceiver.cs

The thing is the .NET class doesn't have any of the methods like in Java to extend from. It calls the ParsePushService, and probably handle the receipt of notification over there, but I don't really know further how to deal with that.

Can you please confirm that there's a way to customize the way remote notification is received on Android, with Xamarin?
